ssl/statem/extensions_clnt.c

```diff
@@ -82,6 +129,7 @@ EXT_RETURN tls_construct_ctos_server_name(SSL_CONNECTION *s, WPACKET *pkt,
     }
```

Comment:
It seems we have some duplicated code here between the no-ech and ech block, i.e. all the WPACKET stuff.  It seems there is an opportunity to improve this.

Reply:
Agreed, it was done that way to make rebasing cleaner and less troublesome but now we're looking at merging, some small changes to the no-ech block would make this all better. I'm not sure if/when it's best to make such changes but will do this one now then can learn what's best as we go.